plugins/net_plugin/net_plugin.cpp

```diff
@@ -2768,6 +2776,16 @@ namespace eosio {
                visitors < connections.get_max_client_count())) {
             fc_ilog(logger, "Accepted new connection: " + paddr_str);
 
+            connections.for_each_supplied_peer([&listen_address, &paddr_str, &limit](const string& peer_addr) {
+               auto [host, port, type] = split_host_port_type(peer_addr);
```

Comment:
The list of `host` strings for comparison with `paddr_str` should be calculated only when `supplied_peers` is modified. Then that new list of host strings can be used here instead of doing this work on every new session. Also you can break out of the for each when a match is found.

Reply:
At the moment `supplied_peers` is our source of truth for the list of configured peers.  I didn't want to create a parallel data structure for that with peer discovery likely to make it obsolete.

`std::for_each` has no mechanism to break out.  The return value of the UnaryFunction is ignored.

Reply:
> `std::for_each` has no mechanism to break out. The return value of the UnaryFunction is ignored

Can you use `std::any_of`?

Reply:
Yes.  Switched to `std::any_of`.